File: string/Solution541.java
class Solution541 {
    public String reverseStr(String s, int k) {
    	char[] ele = s.toCharArray();
    	for(int i = 0; i < s.length(); i += 2 * k){
    		int left = i; 
    		int right = i + k - 1;
    		if(i + k - 1 >= s.length()){
    			right = s.length() - 1;
    		}
    		while(right > left){
    			char cur = ele[left];
    			ele[left] = ele[right];
    			ele[right] = cur;
    			left += 1;
    			right -= 1;
    		}
    	}
    	return String.valueOf(ele);
    }

    public static void main(String[] args){
    	Solution541 s = new Solution541();
    	System.out.println(s.reverseStr("abcdefg", 2));
    	System.out.println(s.reverseStr("abcdefg", 10));

    }
}

File: C++/Solution541.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<queue>
using namespace std;

typedef long long ll;
typedef pair<int, int> pii;

class Solution541 {
public:
    string reverseStr(string s, int k) {
        string ans;
        for(int i = 0; i < s.length(); i += 2 * k){
            if(s.length() - i <= k){
                ans += reverse(s.substr(i, s.length() - i));
            }
            else if(s.length() - i > k && s.length() - i <= 2 * k){
                ans += reverse(s.substr(i, k));
                ans += s.substr(i + k, s.length() - (i + k));
            }
            else{
                ans += reverse(s.substr(i, k));
                ans += s.substr(i + k, k);
            }
        }
        return ans;
    }

    string reverse(string s){
        string ans;
        for(auto ele: s){
            ans = ele + ans;
        }
        return ans;
    }
};

int main(){
    Solution541 s;
    string str = "abcdefg";
    cout << s.reverseStr(str, 2) << endl;
    return 0;
}